chain/client/src/client.rs

```diff
@@ -795,8 +799,9 @@ impl Client {
                     return Err(Error::Other("Invalid chunk version".to_string()));
                 };
 
+                let pec = PartialEncodedChunk::clone(&partial_encoded_chunk);
```

Comment:
not related to this PR per se, but I wonder whether we should clone here. It seems that there are often cases where `process_partial_encoded_chunk` returns early because we already know this partial encoded chunk and therefore there is no need to take ownership here. Maybe passing it as a reference to `ShardsManager::process_partial_encoded_chunk` would be better

Reply:
Yes, I agree this clone seems wasteful. Do you want me to make that change here or file an issue and do it in a separate PR?

Reply:
Sure let's file it as a separate issue

Reply:
https://github.com/near/nearcore/issues/3700